pkg/declcfg/declcfg_to_model.go

```diff
@@ -0,0 +1,108 @@
+package declcfg
+
+import (
+	"fmt"
+
+	"github.com/operator-framework/operator-registry/pkg/model"
+	"github.com/operator-framework/operator-registry/pkg/property"
+)
+
+func ConvertToModel(cfg DeclarativeConfig) (model.Model, error) {
+	mpkgs := model.Model{}
+	defaultChannels := map[string]string{}
+	for _, p := range cfg.Packages {
+		mpkg := &model.Package{
+			Name:        p.Name,
+			Description: p.Description,
+			Channels:    map[string]*model.Channel{},
+		}
+		if p.Icon != nil {
```

Comment:
I don't think I understand your suggestion and how it relates to this particular piece of code.

In general the answer to your question is:
1. We persist icons in `olm.package` blobs in an optional first class field called `icon`. This field contains a base64-encoded `data` value and a mediatype `value` that specifies the format of the data.
2. The icon is in the `olm.package` schema (rather than `olm.bundle`) to reduce unnecessary duplication of the same large icon data blob in all of the `olm.bundle` blobs.

If that's what you're looking for, it seems more applicable as GoDoc comments on the `declcfg.Package` or `declcfg.Icon` types. WDYT?